api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/context/MessageContextLoader.java

```diff
@@ -127,6 +127,20 @@ public ReportClaimMessageContext loadReportClaimMessageContext(Message message)
                 .build();
     }
 
+    public ReportPaymentMessageContext loadReportPaymentMessageContext(Message message) {
+        ReportPaymentMessagePayload payload = payloadMapper.getPayload(message, ReportPaymentMessagePayload.class);
+        Claim claim = getAndCheckClaim(payload.getClaimId());
+        Optional<PaymentCycle> paymentCycle = payload.getPaymentCycleId().flatMap(id -> paymentCycleRepository.findById(id));
```

Comment:
Why does this need to complexity of `flatMap` when you could just do `paymentCycleRepository.findById(payload.getPaymentCycleId())` and no `flatMap` required?

Reply:
Because `payload.getPaymentCycleId()` returns an optional. The alternative would be something like
```
Optional<PaymentCycle> paymentCycle = payload.getPaymentCycleId().ifPresentOrElse(id -> paymentCycleRepository.findById(id), Optional.empty())
```

Reply:
Yep, think I prefer `ifPresentOrElse`.

Reply:
As discussed, sticking with flatMap.